### PR TITLE
Add uproot-raw codegen

### PR DESCRIPTION
**Pull request by ponyisi**
Add the uproot-raw codegen to the standard ServiceX codebase. Supports multitree functionality.

### CODE REVIEW

File: code_generator_raw_uproot/transformer_capabilities.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "Uproot transformer using native uproot arguments",
+  "description": "Extracts data from flat ntuple style root files.",
+  "limitations": "Would be good to note what isn't implemented",
+  "file-formats": ["parquet"],
```
Comment (BenGalewsky) [inline code_generator_raw_uproot/transformer_capabilities.json:5]: Need to indicate the native ability to write root files too. See example in [python code generator](https://github.com/ssl-hep/ServiceX/blob/d92749f04d35035c51ccf4900e9ee1122daa6e96/code_generator_python/transformer_capabilities.json#L5)

File: code_generator_raw_uproot/tests/test_src.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+def test_dummy():
```
Comment (BenGalewsky) [inline code_generator_raw_uproot/tests/test_src.py]: See the python code generator for examples of how to write unit tests for code generators...

Would be good to test the various ways it throws `GenerateCodeException`s

**Review by BenGalewsky — CHANGES_REQUESTED** [review]
This will be interesting to see how it will be used! Just a couple of fixes needed

File: code_generator_raw_uproot/tests/test_src.py (hunk 2 above)
Comment (ponyisi) [reply code_generator_raw_uproot/tests/test_src.py]: Done

File: code_generator_raw_uproot/transformer_capabilities.json (hunk 1 above)
Comment (ponyisi) [reply code_generator_raw_uproot/transformer_capabilities.json:5]: Done

**Review by BenGalewsky — APPROVED** [review]
Ok - I can't wait to see how this pans out in real-world use!

**Comment by gordonwatts** [comment]
Great to see this go in - but also not totally comfortable with the idea we need to put all code generators in this main package. :-)

**Comment by ponyisi** [comment]
@gordonwatts - we don't need them to be - right now I run the uproot-raw codegen in testing4 from a private docker repo. There's a separate question about the frontend interface, which I think I will go some distance to fixing in another PR.

**Pull request merged**